src/mintermization.cc

```diff
@@ -115,37 +115,30 @@ void Mata::Mintermization::trans_to_bdd_afa(const IntermediateAut &aut)
             if (bdd.val.IsZero())
                 continue;
             trans_to_bddvar[ds_pair.first] = bdd.val;
-            bdds.push_back(bdd.val);
+            bdds.insert(bdd.val);
         }
     }
 }
 
-std::vector<BDD> Mata::Mintermization::compute_minterms(const std::vector<BDD>& bdds)
+std::unordered_set<BDD> Mata::Mintermization::compute_minterms(const std::unordered_set<BDD>& bdds)
 {
-    std::vector<BDD> stack;
-    // we start mintermization with first bdd and its negation
-    stack.push_back(bdds.front());
-    stack.push_back(!bdds.front());
```

Comment:
What about the mentioned negation? Shouldn't it be added as well? Or is it included in the `bdd_mng.bddOne()`?

Reply:
BDD at front is now processed in the following loop where we do both `1 && front` and `1 && !front`, so we add it to the set of minterms